repair/repair.cc

```diff
@@ -1342,17 +1342,16 @@ future<> repair_service::bootstrap_with_repair(locator::token_metadata_ptr tmptr
     using inet_address = gms::inet_address;
     return seastar::async([this, tmptr = std::move(tmptr), tokens = std::move(bootstrap_tokens)] () mutable {
         seastar::sharded<replica::database>& db = get_db();
-        auto keyspaces = db.local().get_non_local_strategy_keyspaces();
+        auto ks_erms = db.local().get_non_local_strategy_keyspaces_erms();
         auto myip = utils::fb_utilities::get_broadcast_address();
```

Comment:
Very good.

Idea: add bool effective_replication_map::outdated() const that becomes true if replication changed, so we can abort repair. Soon we'll start waiting for all users of an old topology to complete before moving to a new topology, and this will allow us to move quicker.

Reply:
sounds good. we should be able to mark the e_r_m as outdated in O(1) when it's replaced in the registry.